pymc3/tests/test_distributions.py

```diff
@@ -553,6 +553,30 @@ def RandomPdMatrix(n):
     return np.dot(A, A.T) + n * np.identity(n)
 
 
+def test_hierarchical_logpt():
+    """Make sure there are no random variables in a model's log-likelihood graph."""
+    with pm.Model() as m:
+        x = pm.Uniform("x", lower=0, upper=1)
+        y = pm.Uniform("y", lower=0, upper=x)
+
+    logpt_ancestors = list(ancestors([m.logpt]))
+    ancestors_with_owner = [a for a in logpt_ancestors if a.owner]
+    assert len(ancestors_with_owner) > 0
+    assert not any(isinstance(v.owner.op, RandomVariable) for v in ancestors_with_owner)
+    assert x.tag.value_var in logpt_ancestors
+    assert y.tag.value_var in logpt_ancestors
```

Comment:
Would this be a good place to mock the respective `transform` function and do an `assert_called_once_with`?

Reply:
Are we concerned about the number of times a transform is called?

Reply:
I guess not really, I just wanted to suggest a more targeted way to test the behavior.